nncf/experimental/post_training/compression_builder.py

```diff
@@ -0,0 +1,94 @@
+"""
+ Copyright (c) 2022 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from typing import TypeVar
+
+from nncf.common.utils.logger import logger as nncf_logger
+
+from nncf.experimental.post_training.api.engine import Engine
+from nncf.experimental.post_training.api.dataloader import DataLoader
+from nncf.experimental.post_training.algorithms import Algorithm
+
+from nncf.experimental.post_training.backend import Backend
+from nncf.experimental.post_training.backend import get_model_backend
+
+ModelType = TypeVar('ModelType')
+
+
+class CompressionBuilder:
+    """
+    The main class applies the compression algorithms to the model according to their order.
+    """
+
+    def __init__(self):
+        self.algorithms = []
+
+    def add_algorithm(self, algorithm: Algorithm) -> None:
+        """
+        Adds the algorithm to the pipeline.
+        """
+        self.algorithms.append(algorithm)
+
+    def _create_engine(self, backend: Backend) -> Engine:
+        if backend == Backend.ONNX:
+            from nncf.experimental.onnx.engine import ONNXEngine
+            return ONNXEngine()
+        return None
+
+    def _create_statistics_aggregator(self, engine: Engine, dataloader: DataLoader, backend: Backend):
+        if backend == Backend.ONNX:
+            from nncf.experimental.onnx.statistics.aggregator import ONNXStatisticsAggregator
+            return ONNXStatisticsAggregator(engine, dataloader)
+        return None
+
+    def _get_prepared_model_for_compression(self, model: ModelType, backend: Backend) -> ModelType:
+        if backend == Backend.ONNX:
+            from nncf.experimental.onnx.utils import modify_onnx_model_for_quantization
+            return modify_onnx_model_for_quantization(model)
+        return None
```

Comment:
Sorry for new comment. I just realized that this code is not flexible enough for other backends.
I believe that writing a new if-statement in multiple places for the new backend is not what we want.
What about introducing a registry of backends and a class encapsulating backend specific objects, roughly like this: 
```python
class BaseBackendHandler:
  def get_engine()
  def get_prepared_model_for_compression()
  def create_stats_agregator()

@BACKEND_HANDLER.register(Backend.ONNX)
class ONNXBackendHandler(BaseBackendHandler)

```

Reply:
This is a good improvement. I think that we could do it in the future PRs, not in this one. The main idea of this PR is to add a minimal functional MinMaxQuantization for ONNX.